iceoryx_utils/test/moduletests/test_index_queue.cpp

```diff
@@ -220,4 +220,82 @@ TYPED_TEST(IndexQueueTest, popIfFullReturnsNothingWhenQueueIsNotFull)
     EXPECT_FALSE(q.popIfFull().has_value());
 }
 
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastReturnsNothingIfQueueIsEmpty)
+{
+    auto& q = this->queue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(1).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueIsFull)
+{
+    auto& q = this->fullQueue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(0).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueIsEmpty)
+{
+    auto& q = this->queue;
+    EXPECT_FALSE(q.popIfSizeIsAtLeast(0).has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastZeroReturnsNothingIfQueueContainsOneElement)
+{
+    auto& q = this->queue;
+    auto index = this->fullQueue.pop();
+    ASSERT_TRUE(index.has_value());
+    q.push(*index);
+
+    index = q.popIfSizeIsAtLeast(0u);
+    ASSERT_FALSE(index.has_value());
+}
+
+TYPED_TEST(IndexQueueTest, popIfSizeIsAtLeastOneReturnsIndexIfQueueContainsOneElement)
+{
+    auto& q = this->queue;
+    using index_t = typename TestFixture::index_t;
+    auto index = this->fullQueue.pop();
+    ASSERT_TRUE(index.has_value());
+
+    // we want to get the last index (to push it into q)
+    auto index1 = this->fullQueue.pop();
+    while (index1.has_value())
+    {
+        index = std::move(index1);
+        index1 = this->fullQueue.pop();
+    }
+
+    index_t expectedIndex = *index;
+    q.push(*index);
```

Comment:
A leftover from the time were indices could only be produced by the queue itself. Can be simplified now.

Reply:
Simplified it somewhat, but it is not as simple as your example makes it seem to be. The index queue can only deal with indices between 0 and capacity-1, by construction. Since this are typed tests, the capacity is sometimes lower than a fixed constant. I changed it to push capacity-1, which is always valid.